src/context.ts

```diff
@@ -12,19 +12,40 @@ export default class Context {
    * B. Saved to disk if in record mode
    */
   public interceptedRequestsCompleted: ISerializedHttp[] = [];
-  /**
-   * Serialized records loaded from disk.
-   */
-  public loadedMocks: ISerializedHttp[] = [];
   /**
    * Proxied requests which have not yet responded. When completed
    * the value is set to "null" but the index is preserved.
    */
   public inFlightRequests: Array<IInFlightRequest | null> = [];
 
+  /**
+   * Serialized records loaded from disk.
+   */
+  private mocks: ISerializedHttp[] = [];
+  /**
+   * Mocks which have already been consumed.  The ordinal position of
+   * each entry is a boolean that, when true, indicates the parallel
+   * array entry in loadedMocks has been consumed.
+   */
+  private consumed: boolean[] = [];
+
   public clear() {
     this.interceptedRequestsCompleted = [];
     this.inFlightRequests = [];
     this.loadedMocks = [];
   }
+
+  public get consumedMocks(): boolean[] {
+    return this.consumed;
+  }
+
+  public get loadedMocks(): ISerializedHttp[] {
```

Comment:
Made this a property accessor so that the consumer doesn't need to care about keeping the "consumedMocks" array in sync on changes.